### PR TITLE
DP 603: Remove api_sql_call

### PR DESCRIPTION
**Pull request by jason-p-pickering**
### Summary of Proposed Changes

- Removes api_sql_call from datapackr
- Replaced with equivalent function getSQLView from datimutils

### Related Issues
- DP-603

## Reviewer:
- [ ] Tests added/updated & passing.
- [ ] Clean linting.
- [ ] Related issue ticket in Jira/GitHub.
- [ ] Documentation added/updated.
- [ ] Code conforms to style guidelines.
- [ ] Well commented.
- [ ] Updates reflected in NEWS.md.
- [ ] Build check passes.

### CODE REVIEW

**Comment by jason-p-pickering** [comment]
@cnemarich @jason-p-pickering I just noticed there is a parameter in getCodeList ( aka pullFullCodeList) called `expanded`. 

```
#' @param expanded If TRUE, will add dataset, period, period_dataset, and
#' targets_results, related categoryOption metadata,  as additional columns
```

Looking at the master version of the code, this parameter appears to do nothing in the function. Is it needed?

**Comment by cnemarich** [comment]
> @cnemarich @jacksonsj I just noticed there is a parameter in getCodeList ( aka pullFullCodeList) called `expanded`.
> 
> ```
> #' @param expanded If TRUE, will add dataset, period, period_dataset, and
> #' targets_results, related categoryOption metadata,  as additional columns
> ```
> 
> Looking at the master version of the code, this parameter appears to do nothing in the function. Is it needed?

It looks like @jacksonsj added that parameter in this commit but then no functionality was ever implemented around it. Scott, is this needed?

https://github.com/pepfar-datim/datapackr/commit/9ca0e4b0881970af6cab99110198027daf6230de

File: data-raw/COP21_Data_Pack_processing_script.R [hunk 1]
```diff
@@ -143,7 +143,7 @@ d <- writePSNUxIM(d, snuxim_model_data_path, output_folder)
                         values_drop_na = TRUE) %>%
     dplyr::mutate(support_type = toupper(support_type))
 
-  fullCodeList <- pullFullCodeList(FY = 2022,
+  fullCodeList <- getCodeList(cop_year = 2021,
```
Comment (JordanBalesBAO) [inline data-raw/COP21_Data_Pack_processing_script.R:146]: Do we have / need to create a COP22_Data_Pack_processing_script? Essentially do we know why this was changed to 2022 before & is it noted somewhere?

File: data-raw/update_cached_de_coc_co_map.R [hunk 2]
```diff
@@ -9,16 +9,16 @@ datimutils::loginToDATIM(secrets)
 cop_year = getCurrentCOPYear()
 
 # Pull Code Lists for FY22 data (MER Targets, IMPATT, SUBNAT Targets) ####
-  mer.T <- pullFullCodeList(FY = cop_year +1, datastreams = c("mer_targets"))
-  subnat.T <- pullFullCodeList(FY = cop_year +1, datastreams = c("subnat_targets"))
-  impatt.T <- pullFullCodeList(FY = cop_year +1, datastreams = c("impatt"))
+  mer.T <- getCodeList(cop_year = cop_year, datastreams = c("mer_targets"))
```
Comment (JordanBalesBAO) [inline data-raw/update_cached_de_coc_co_map.R:12]: Why did we change the cop year math for most of these? I presume getCodeList() is the reason, but want to confirm.

**Review by JordanBalesBAO — APPROVED** [review]
Left some clarification questions in regards to 'cop year', but everything else looks fine to me

File: data-raw/COP21_Data_Pack_processing_script.R (hunk 1 above)
Comment (cnemarich) [reply data-raw/COP21_Data_Pack_processing_script.R:146]: This was just a result of flipping from `FY` to `cop_year`.

**Comment by cnemarich** [comment]
@jacksonsj to review question marked above

**Review by jacksonsj — APPROVED** [review]
All good.

**Pull request merged**